### PR TITLE
Add metadata migration setup

### PR DESCRIPTION
**Pull request by Weiko**
## Context
This PR removes the "synchronise" mode from typeorm and create migrations for the metadata tables. I rebased it from the changes made by @magrinj in this PR https://github.com/twentyhq/twenty/pull/1576
I also introduced new columns: display_name, target_table_name, target_object_name.
This is because we will need to dissociate what users will see (display name) and the names of our entities (and their fields).

Our plan is to be have 
```
metadata: {
   Company: {
      url: 'twenty.com'
   }
}
```
mapped to
```
workspace_schema: {
   'companies': {
      'field_1': 'twenty.com
   }
}
```
This will allow us to let users edit field names without creating a new migration (which would have been the case with the previous synchronise mode with 1:1 map between metadata.object_metadata.name and workspace_123.name, same for fields)

Also, this PR adds a first version of our migration generator. Once a field has been added/edited/removed from the metadata table, we compute a migration dynamically (this part is not done yet) and we store the migration in the tenant/workspace schema. Then we ask typeorm to execute it after parsing it.

### CODE REVIEW

File: server/src/tenant/metadata/data-source/data-source.service.ts [hunk 1]
```diff
@@ -39,7 +41,15 @@ export class DataSourceService implements OnModuleInit, OnModuleDestroy {
...
     const queryRunner = this.mainDataSource.createQueryRunner();
+    const schemaAlreadyExists = await queryRunner.hasSchema(schemaName);
+    if (schemaAlreadyExists) {
+      throw new Error(
+        `Schema ${schemaName} already exists for workspace ${workspaceId}`,
+      );
+    }
+
     await queryRunner.createSchema(schemaName, true);
+    await this.createMigrationTable(queryRunner, schemaName);
```
Comment (Weiko) [inline server/src/tenant/metadata/data-source/data-source.service.ts:52]: createMigrationTables

File: server/src/tenant/metadata/data-source/data-source.service.ts [hunk 2]
```diff
@@ -48,6 +58,40 @@ export class DataSourceService implements OnModuleInit, OnModuleDestroy {
     );
   }
 
+  private async createMigrationTable(
+    queryRunner: QueryRunner,
+    schemaName: string,
+  ) {
+    await queryRunner.createTable(
+      new Table({
```
Comment (Weiko) [inline server/src/tenant/metadata/data-source/data-source.service.ts:66]: Note: Ideally, this should be based on server/src/tenant/metadata/migration-generator/tenant-migration.entity.ts but I couldn't make it work

File: server/src/tenant/metadata/data-source/data-source.service.ts (hunk 1 above)
Comment (Weiko) [inline server/src/tenant/metadata/data-source/data-source.service.ts:52]:
```suggestion
    await this.createMigrationTables(queryRunner, schemaName);
```

**Review by charlesBochet — APPROVED** [review]
Let's merge and see! Big one!

**Pull request merged**